src/pages/_app.js

```diff
@@ -1,27 +1,84 @@
 import Layout from "../components/Layout";
 import GlobalStyle from "../styles";
 import uniqueCities from "../utils/getCities";
-import sortedEvents from "../utils/sortEvents";
+import { getSortedEvents, getEventsOfCity } from "../utils/getEvents";
+import { events } from "../lib/data";
 import { useFilterStore } from "../store";
+import { useEffect, useState } from "react";
+import useSWR, { SWRConfig } from "swr";
+
+const BASE_URL = "https://nominatim.openstreetmap.org/reverse?format=jsonv2";
+
+const fetcher = async (url) => {
+  const response = await fetch(url);
+
+  if (!response.ok) {
+    const error = new Error("An error occurred while fetching data.");
+    error.info = await response.json();
+    error.status = response.status;
+    throw error;
+  }
+
+  return response.json();
+};
 
 export default function App({ Component, pageProps }) {
-  const city = useFilterStore((state) => state.city);
+  const [userLocation, setUserLocation] = useState([]);
+  const { city, setCity, currentLocation } = useFilterStore((state) => state);
+
+  const sortedEventsOfCity = getEventsOfCity(city, getSortedEvents(events));
+
+  const { data } = useSWR(
+    userLocation && currentLocation
+      ? `${BASE_URL}&lat=${userLocation[0]}&lon=${userLocation[1]}&zoom=10`
+      : null,
+    fetcher,
+    {
+      refreshInterval: 600000,
+    }
+  );
+
+  useEffect(() => {
+    const options = {
+      enableHighAccuracy: true,
+      timeout: 10000,
+    };
+    const successCallback = ({ coords }) => {
+      const [latitude, longitude] = [coords.latitude, coords.longitude];
+      setUserLocation([latitude, longitude]);
+    };
+    const errorCallback = (error) => {
+      console.log(error);
+    };
+    if (currentLocation) {
+      if (window.navigator.geolocation) {
+        window.navigator.geolocation.getCurrentPosition(
+          successCallback,
+          errorCallback,
+          options
+        );
+      }
+    }
+  }, [userLocation, currentLocation]);
 
-  const events = city
-    ? sortedEvents.filter((event) => {
-        const addressArray = event.location.address.split(",");
-        const postalCity = addressArray[addressArray.length - 2];
-        const cityString = postalCity?.replace(/\d+/, "").trim();
-        return cityString === city;
-      })
-    : sortedEvents;
+  useEffect(() => {
+    if (currentLocation) {
+      data ? setCity(data.address.city) : null;
+    }
+  }, [currentLocation, data, setCity]);
```

Comment:
Please explain line 64-65🙏🏼

Do you want to call setCity() when currentLocation, data and setCtiy change?

Does it make sense to put setCity in the dependency? If yes, why?

When does data become nullish? If data never becomes nullish the ternary operator is unnecessary, right?

Reply:
If the response from the request takes longer than expected if an error occurs `setCity()` would set the City as undefined and the location buttons will get an undefined prop which would make the app crash on the location screen. 
I wrote a lot of error handling so that the buttons only update if the fetch request is successful.